src/extension.ts

```diff
@@ -146,7 +146,10 @@ export const activate = async (context: vscode.ExtensionContext) => {
 
   vscode.workspace.onDidChangeWorkspaceFolders(async (event) => {
     utils.checkForWorkspaceFolders()
-    await loginAndRefresh([...event.added], true)
+    const foldersToRefresh = (event.added.length && event.added) 
+      || vscode.workspace.workspaceFolders 
+      || []
+    await loginAndRefresh([...foldersToRefresh], true)
```

Comment:
do you think it would work if we refreshed the folders from `event.added` and `event.removed`?

Reply:
Seems all the views need is the refresh and refreshing it will make remove the corresponding folder views